llvm/lib/CodeGen/EarlyIfConversion.cpp

```diff
@@ -122,6 +122,17 @@ class SSAIfConv {
   /// The branch condition determined by analyzeBranch.
   SmallVector<MachineOperand, 4> Cond;
 
+  struct PredicationStrategyBase {
```

Comment:
`struct` with inheritance, virtual methods, and `=0`  is really confusing.

Reply:
I agree. Updated to `class` for `PredicationStrategyBase` and its derived classes.